Make resource independent of data bags (#445)
sous-chefs/users: README.md

## Scope

> This cookbook is concerned with the management of OS users and groups. It also manages the distribution of ssh keys to a user's home directory.

Reviewer: I think this might work better.
```suggestion
This cookbook is concerned with the management of OS users and groups (optionally from databags). It also manages the distribution of ssh keys to a user's home directory.
```

> This cookbook is concerned with the management of OS users and groups (optionally from databags). It also manages the distribution of ssh keys to a user's home directory.

Reviewer: You should also update the description on line 9.

## Maintainers

This cookbook is maintained by the Sous Chefs. The Sous Chefs are a community of Chef cookbook maintainers working together to maintain important cookbooks. If you’d like to know more please visit [sous-chefs.org](https://sous-chefs.org/) or come chat with us on the Chef Community Slack in [#sous-chefs](https://chefcommunity.slack.com/messages/C2V7B88SF).

## Requirements

> If you are upgrading from a version < 5.7.0 please see [upgrading.md](https://github.com/sous-chefs/users/upgrading.md)

Reviewer: ```suggestion
If you are upgrading from a version < 6.0.0 please see [upgrading.md](https://github.com/sous-chefs/users/upgrading.md)
```

users_manage 'GROUPNAME' do
  group_id GROUPID
  action [:create]

>   users users_variable

Reviewer: Ideally, it'd be nice to still show examples for both with and without databags to help users.

Author: I am showing three examples. One where I use a data bag, one where I use attributes and one where I use a local variable. So I think it is covered. If not, can you let me know what exampel you want specifically?

Reviewer: Ah yes, I should have looked at the whole file as rendered. Something I think would be helpful here is to show what `users_variable` should look like or just show a simple example without using a variable.

### users_manage

> The `users_manage` resource manages users and groups based off the `users` property and the specified action(s).

Reviewer: ```suggestion
The `users_manage` resource manages users and groups based off the `users` property or of a data bag search and the specified action(s).
```
